packages/builder/src/builderStore/store/frontend.js

```diff
@@ -1040,6 +1042,30 @@ export const getFrontendStore = () => {
           if (component[name] === value) {
             return false
           }
+
+          const settings = getComponentSettings(component._component)
+          const updatedSetting = settings.find(setting => setting.key === name)
+
+          if (
+            updatedSetting.type === "dataSource" ||
+            updatedSetting.type === "table"
+          ) {
+            const { list: allTables } = get(tables)
+            const { schema } = allTables.find(
```

Comment:
Settings of type `dataSource` don't need to be tables, so searching the tables list here will return `undefined` for any non-table datasource. That mean's we won't update the schema for these (which is probably fine) but it will throw an error here when it attempt to destructure out `schema` of `undefined.

A better thing to do here would actually be using the existing `getSchemaForDatasource` function in `dataBinding.js`, and just pass in the current value. This will get the correct schema for any type of data source!

We don't need to use this up above at component creation since we know that all tables have a `schema` property directly on them, so we can just use that.